tree/dataframe/inc/ROOT/RDF/PyROOTHelpers.hxx

```diff
@@ -69,6 +70,31 @@ ROOT::RDF::RResultPtr<std::vector<T>> RDataFrameTake(ROOT::RDF::RNode df, std::s
    return df.Take<T>(column);
 }
 
+// MakeRDataFrame helpers
+
+// Dummy RResultPtr class so that we can reuse RLazyDS for reading numpy arrays
+template <typename T>
+class RResultPtrDummy {
+private:
+   T* fPtr;
+public:
+   using Value_t = T;
+   RResultPtrDummy(T& obj) : fPtr(&obj) {};
+   T *operator->() { return fPtr; }
+   T &operator*() { return *fPtr; }
+};
+
+// MakeLazyDataFrame variant returning an RDataFrame on the heap so that we can
+// build a Python proxy for it.
+template <typename... Columns>
+RDataFrame* NewLazyDataFrame(std::pair<std::string, Columns> &&... colNameProxyPairs)
+{
+   return new RDataFrame(std::make_unique<RLazyDS<Columns...>>(
+      std::forward<std::pair<std::string, Columns>>(colNameProxyPairs)...));
+}
```

Comment:
We could either avoid duplicating `MakeLazyDataFrame` just to have an RDF on the heap by using the following instead:
```c++
new RInterface<RLoopManager>(MakeLazyDataFrame(...))
```

Reply:
Is the type still `ROOT::RDataFrame`? If not, this could have some impliciations with cppyy...

Reply:
no, the type is `RInterface<RLoopManager>`

Reply:
in case it is not possible to reuse `MakeLazyDataFrame` please add a comment explaining why we _need_ a function that returns exactly a `RDataFrame` and why it must be on the heap